internal/services/monitor/monitor_alert_prometheus_rule_group_resource.go

```diff
@@ -0,0 +1,571 @@
+				return fmt.Errorf("deleting %s: %+v", id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func expandPrometheusRuleModel(inputList []PrometheusRuleModel, d *schema.ResourceData) []prometheusrulegroups.PrometheusRule {
+	outputList := make([]prometheusrulegroups.PrometheusRule, 0)
+
+	for i, v := range inputList {
+		output := prometheusrulegroups.PrometheusRule{
+			Enabled:    pointer.To(v.Enabled),
+			Expression: v.Expression,
+			Labels:     pointer.To(v.Labels),
+		}
+
+		if v.Alert != "" {
+			output.Actions = expandPrometheusRuleGroupActionModel(v.Action)
+			output.Alert = pointer.To(v.Alert)
+			if _, ok := d.GetOk("rule." + strconv.Itoa(i) + ".severity"); ok {
+				output.Severity = pointer.To(v.Severity)
+			}
+			output.Annotations = pointer.To(v.Annotations)
+			output.For = pointer.To(v.For)
+			output.ResolveConfiguration = expandPrometheusRuleResolveConfigurationModel(v.ResolveConfiguration)
+		} else {
+			// action, alert, severity, annotations, for, resolve_configuration must be empty when type is recording rule
+			output.Record = pointer.To(v.Record)
+		}
+
+		outputList = append(outputList, output)
+	}
+
+	return outputList
+}
+
+func expandPrometheusRuleGroupActionModel(inputList []PrometheusRuleGroupActionModel) *[]prometheusrulegroups.PrometheusRuleGroupAction {
+	outputList := make([]prometheusrulegroups.PrometheusRuleGroupAction, 0)
+	for _, v := range inputList {
+		output := prometheusrulegroups.PrometheusRuleGroupAction{
+			ActionProperties: pointer.To(v.ActionProperties),
+		}
+
+		output.ActionGroupId = pointer.To(v.ActionGroupId)
+
+		outputList = append(outputList, output)
+	}
+
+	return &outputList
+}
+
+func expandPrometheusRuleResolveConfigurationModel(inputList []PrometheusRuleResolveConfigurationModel) *prometheusrulegroups.PrometheusRuleResolveConfiguration {
+	if len(inputList) == 0 {
+		return nil
+	}
+
+	input := &inputList[0]
+	output := prometheusrulegroups.PrometheusRuleResolveConfiguration{
+		AutoResolved: pointer.To(input.AutoResolved),
+	}
+
+	output.TimeToResolve = pointer.To(input.TimeToResolve)
+
+	return &output
+}
+
+func flattenPrometheusRuleModel(inputList *[]prometheusrulegroups.PrometheusRule) []PrometheusRuleModel {
+	outputList := make([]PrometheusRuleModel, 0)
+	if inputList == nil {
+		return outputList
+	}
+
+	for _, input := range *inputList {
+		output := PrometheusRuleModel{
+			Expression: input.Expression,
+		}
+
+		actionsValue := flattenPrometheusRuleGroupActionModel(input.Actions)
+
+		output.Action = actionsValue
+
+		output.Alert = pointer.From(input.Alert)
+
+		output.Annotations = pointer.From(input.Annotations)
+
+		output.Enabled = pointer.From(input.Enabled)
+
+		output.For = pointer.From(input.For)
+
+		output.Labels = pointer.From(input.Labels)
+
+		output.Record = pointer.From(input.Record)
+
+		resolveConfigurationValue := flattenPrometheusRuleResolveConfigurationModel(input.ResolveConfiguration)
+
+		output.ResolveConfiguration = resolveConfigurationValue
+
```

Comment:
we don't need these blank lines?
```suggestion
		output.Alert = pointer.From(input.Alert)
		output.Annotations = pointer.From(input.Annotations)
		output.Enabled = pointer.From(input.Enabled)
		output.For = pointer.From(input.For)
		output.Labels = pointer.From(input.Labels)
		output.Record = pointer.From(input.Record)
		resolveConfigurationValue := flattenPrometheusRuleResolveConfigurationModel(input.ResolveConfiguration)
		output.ResolveConfiguration = resolveConfigurationValue
```

Reply:
Thanks for your feedback. I have updated the code. Could you please take another look?